widget/embedded/src/components/SwapDetailsModal/SwapDetailsModal.WalletState.tsx

```diff
@@ -1,30 +1,33 @@
 import type { WalletStateContentProps } from './SwapDetailsModal.types';
 
-import { MessageBox, Wallet, WalletState } from '@rango-dev/ui';
+import { MessageBox, Wallet } from '@rango-dev/ui';
 import { useWallets } from '@rango-dev/wallets-react';
 import React from 'react';
 
 import { getContainer } from '../../utils/common';
+import { mapStatusToWalletState } from '../../utils/wallets';
 
 import { WalletContainer } from './SwapDetailsModal.styles';
 
 export const WalletStateContent = (props: WalletStateContentProps) => {
   const { type, title, currentStepWallet, message, showWalletButton } = props;
-  const { connect, getWalletInfo, state: walletState } = useWallets();
+  const { connect, getWalletInfo, state } = useWallets();
   const walletType = currentStepWallet?.walletType;
-  const isConnected = walletType && walletState(walletType).connected;
-  const state = isConnected ? WalletState.CONNECTED : WalletState.DISCONNECTED;
+  const walletState = !!walletType && mapStatusToWalletState(state(walletType));
+  const walletInfo = !!walletType && getWalletInfo(walletType);
+  const shouldShowWallet = showWalletButton && walletState && walletInfo;
```

Comment:
I think `shouldShowWallet` should be a `boolean`, but it's currently inferred as `false | WalletInfo`. Maybe is better to remove !! from `walletType` and place them before `walletState` && `walletInfo`. What do you think?